### ISSUE TITLE
Block ads and tracker run time

### ISSUE THREAD

**Issue by vickyrathee** [cliqz-oss/adblocker#2223]
I have easylist downloaded using a CRON and parsing it using `PuppeteerBlocker.parse`

```
const blocker = PuppeteerBlocker.parse([
  fs.readFileSync('easylist.txt', 'utf8'),
  fs.readFileSync('easyprivacy.txt', 'utf8'),
].join('\n'));
```

In my app, I have option to let users set if the ads and tracker should be blocked or not

```
export interface BlockingOptions {
  blockAds : boolean = true
  blockTracker: boolean = false
}
````

Can I specify which list to apply dynamically? Or I should create 2 instance of `blocker ` and apply with if/else individually to use `easylist.txt` if adsBlock is true; and  `easyprivacy.txt` if blockTracker is true; or both if both are true?

Or is there anything I can see from `blocker.match()` return data to see, where it matched from. easylist or easyprivacy to use the combined list?

```
page.on("request", (req) => {
      if(options.blockAds){
        const { redirect, match } = blocker.match(fromPuppeteerDetails(req));
        if (redirect !== undefined) {
          return req.respond({
            body: redirect.body,
            contentType: redirect.contentType,
          });
        }        
        if(match == true){
          return req.abort('blockedbyclient');
        }      
     }
     return req.continue();
});
```

**Comment by remusao** [comment]
Hi @vickyrathee,

At the moment I think there is no way to selectively enable part of the filters used in an instance of blocker, since all filters are added at once without knowledge of origin. I can see two options:

1. Create different instances of the blocker, either every time the user selects different settings (e.g. toggling blocking tracker or just ads), or by serializing them on disk for fast access with `const serialized = blocker.serialize()` (this returns an UInt8Array which can be stored on disk then restored with `PuppeteerBlocker.deserialize(serialized)` (and this is *very* fast).
2. You can keep a mapping from each filter ID to its origin list but this requires a bit of work on your side as you will have to first parse the list, then create the mapping, and only once that is done you can create the instance of blocker. Then in your `request` listener, on match, you can get the ID of the filter which triggered, and look it up in your mapping to know from which list it originates.

If you decide to go to 2. then I can provide more info as to how this could work.

Best,

**Comment by vickyrathee** [comment]
@remusao I think I will go with 1st, as that's seems easy to implement and will improve the performance with if/else to dynamically apply the serialized `blocker` or `tracker` or both.

**Comment by remusao** [comment]
Closing this one. Feel free to open another issue if you need anything else.
